File: Java/1557. Minimum Number of Vertices to Reach All Nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        List<Integer> ans=new ArrayList<>();
        boolean[] check=new boolean[n];
        for (List<Integer> e: edges){
            check[e.get(1)]=true;
        }
        for (int i = 0; i < n; ++i){
            if (check[i] == false){
                ans.add(i);
            }
        }
        return ans;
    }
}

File: C++/1557. Minimum Number of Vertices to Reach All Nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> ans;
        vector<int> check(n);
        for (auto e: edges){
            check[e[1]] = 1;
        }
        for (int i = 0; i < n; ++i){
            if (check[i] == 0){
                ans.push_back(i);
            }
        }
        return ans;
    }
};
